internal/telemetry/telemetry.go

```diff
@@ -0,0 +1,92 @@
+// Copyright 2022 MongoDB Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package telemetry
+
+import (
+	"encoding/json"
+	"errors"
+	"os"
+	"path"
+
+	"github.com/mongodb/mongocli/internal/config"
+	"github.com/spf13/afero"
+)
+
+const (
+	dirPermissions   = 0700
+	filePermissions  = 0600
+	cacheFilename    = "telemetry"
+	maxCacheFileSize = 100 * 1024 * 1024 // 100MB
+)
+
+var fs = afero.NewOsFs()
+
+type Event struct {
+	Timestamp  string            `json:"timestamp"`
+	Source     string            `json:"source"`
+	Name       string            `json:"name"`
+	Properties map[string]string `json:"properties"`
```

Comment:
To match the API this should be 
```suggestion
	Properties map[string]interface{} `json:"properties"`
```

Reply:
OK, I'll change that.